src/main/java/bio/terra/common/ErrorCollector.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.common;
+
+import bio.terra.service.dataset.exception.IngestFailureException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ErrorCollector {
+  private List<String> errors;
+  private int maxBadLoadFileLineErrorsReported;
+  private String exceptionMessage;
+
+  public ErrorCollector(int maxBadLoadFileLineErrorsReported, String exceptionMessage) {
+    this.errors = new ArrayList<>();
+    this.maxBadLoadFileLineErrorsReported = maxBadLoadFileLineErrorsReported;
+    this.exceptionMessage = exceptionMessage;
+  }
+
+  public boolean anyErrorsCollected() {
+    return errors.size() > 0;
+  }
+
+  public void record(String lineErrorMsgFormat, String lineErrorMsg) {
```

Comment:
You could generalize this to accept any args `format()` would allow
```suggestion
  public void record(String lineErrorMsgFormat, Object.. args) {
```

Reply:
Can you explain this suggestion more? I don't understand.

Reply:
The parameters `lineErrorMsgFormat`, `lineErrorMsg` are passed to `String.format()`. The signature of this method is `format(String format, Object... args)`. So declaring `lineErrorMsg` as `Object...` would allow callers to use all the features of `String.format()`, e.g., if they wanted to provide more parameters to the error format string.

Reply:
For example right you have
```
  errorCollector.record("Error at line " + lineCount + ": %s", ex.getMessage())
```

with this change you could instead write
```
  errorCollector.record("Error at line %d: %s", lineCount , ex.getMessage())
```

Reply:
Ahh ok, thanks for the further detail! Should be changed.